rct_optimizations/src/rct_optimizations/covariance_analysis.cpp

```diff
@@ -169,4 +169,118 @@ CovarianceResult computeCovariance(ceres::Problem &problem,
 
   return res;
 }
+
+CovarianceResult computeCovarianceInTangentSpace(ceres::Problem &problem,
```

Comment:
Did you look at using [this function in Ceres](http://ceres-solver.org/nnls_covariance.html#_CPPv4NK5ceres32GetCovarianceBlockInTangentSpaceEPKdPKdPd) to get the covariance block of the tangent space? Seems like it might be equivalent and much easier than adding a whole new function

Reply:
I tried that first but was unsuccessful due to runtime crashing. I assumed it was because the size of the covariance matrix being passed in was not correct, which led me to to believe it was including other parameters that I wanted to ignore so I ended up with this.